### PR TITLE
Add changelog entries

### PR DESCRIPTION
**Pull request by tcharding**
Gearing up for the next release add changelog entries for everything merged upto July 24.

### CODE REVIEW

File: bitcoin/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,50 @@
...
+
+- Removed types:
+  - Remove [reexport of psbt::Prevouts](https://github.com/rust-bitcoin/rust-bitcoin/pull/1872)
+  - [Move witness types to the script module](https://github.com/rust-bitcoin/rust-bitcoin/pull/1846)
+  - [Rename `PartiallySignedTransaction` to `Psbt`](https://github.com/rust-bitcoin/rust-bitcoin/pull/1938)
+
+- Error handling improvements:
+  - [Improve `hashes::Error`](https://github.com/rust-bitcoin/rust-bitcoin/pull/1873)
+  - [Add `ValidationError`](https://github.com/rust-bitcoin/rust-bitcoin/pull/1874)
+  - [Improve `crytpo::taproot` error type](https://github.com/rust-bitcoin/rust-bitcoin/pull/1895)
```
Comment (apoelstra) [inline bitcoin/CHANGELOG.md:39]: In 4c27d1c53e895c68ccf7f9e161e522ae699ab665:

typo `crytpo`

**Comment by apoelstra** [comment]
Otherwise looks good! I did a quick check of the PR list and everything I spot-checked that should be there is there.

**Comment by tcharding** [comment]
Cheers, force pushed typo fix.

**Review by apoelstra — DISMISSED** [review]
ACK ec417200a1aa5b9b6ce1d2b0ff84240bd853ad77

**Review by sanket1729 — APPROVED** [review]
utACK ec417200a1aa5b9b6ce1d2b0ff84240bd853ad77

File: bitcoin/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,46 @@
...
+  - [Add a checked version of weight mul fee_rate](https://github.com/rust-bitcoin/rust-bitcoin/pull/1864)
+  - [Use `Amount` type for `TxOut` value field](https://github.com/rust-bitcoin/rust-bitcoin/pull/1811)
+  - Add support for [alloc-free parse errors](https://github.com/rust-bitcoin/rust-bitcoin/pull/1297)
+  - [Rename `TaprootSpendInfo::as_script_map` `to script_map`](https://github.com/rust-bitcoin/rust-bitcoin/pull/1897)
+  - [Add `Amount::from_int_btc`](https://github.com/rust-bitcoin/rust-bitcoin/pull/1870)
+  - Make [`ckd_priv` function private](https://github.com/rust-bitcoin/rust-bitcoin/pull/1882) and impl `AsRef` for `ChildNumber`.
+  - [Use hex_lit::hex in benches](https://github.com/rust-bitcoin/rust-bitcoin/pull/1941) (backported to v0.30.1)
+  - [Add a verify function to PublicKey](https://github.com/rust-bitcoin/rust-bitcoin/pull/1911)
+  - [script: Add method `count_sigops`](https://github.com/rust-bitcoin/rust-bitcoin/pull/1890)
+    - [Implement `From<PublicKey> for XOnlyPublicKey`](https://github.com/rust-bitcoin/rust-bitcoin/pull/1901)
```
Comment (sanket1729) [inline bitcoin/CHANGELOG.md:26]: nit: Indentation.

Comment (apoelstra) [reply bitcoin/CHANGELOG.md:26]: oh, lol, this does actually affect the rendered output. Good catch.

Comment (tcharding) [reply bitcoin/CHANGELOG.md:26]: Fixed, thanks.

**Review by apoelstra — APPROVED** [review]
ACK 402885f48f71820e4d0f815c54775708a3259b4d

**Comment by tcharding** [comment]
I hadn't reviewed the rendered version of the changelog, did so just now. A bunch of code tick usage is missing - will add and force push. Please don't merge yet.

**Comment by tcharding** [comment]
I've attempted to make the [rendered HTML](https://github.com/tcharding/rust-bitcoin/blob/07-24-changelog/bitcoin/CHANGELOG.md) clearer to read. Please do nitpicky review so I can use this format for good from now.

**Comment by apoelstra** [comment]
Looks great!

**Review by apoelstra — DISMISSED** [review]
ACK aa28e97258ffcdb507fe898da82d2cd2950ec15b

**Comment by yancyribbens** [comment]
ack https://github.com/rust-bitcoin/rust-bitcoin/pull/1952/commits/aa28e97258ffcdb507fe898da82d2cd2950ec15b

**Review by apoelstra — DISMISSED** [review]
ACK 1d1e862f551b25c20f62a2aea1515c19074e7a47

**Comment by tcharding** [comment]
FTR this is not the complete list of changelog entries required for this release. Its just upto July 24, and proves the formatting we want to use.

File: bitcoin/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,49 @@
+# Unreleased (changelog done for PRs merged as of June 24)
+
+- Bump MSRV to Rust 1.48.0 [#1729](https://github.com/rust-bitcoin/rust-bitcoin/pull/1729)
+- Add new example code for signature verification[#1776](https://github.com/rust-bitcoin/rust-bitcoin/pull/1776)
```
Comment (clarkmoody) [inline bitcoin/CHANGELOG.md:4]: Nit here and for the remainder of the entries: need a space between the item and the PR link. The first item has it, but not the rest.

- So this is what we want [#123](https://github.com/rust-bitcoin/rust-bitcoin/pull/123)
- And not this[#456](https://github.com/rust-bitcoin/rust-bitcoin/pull/456)

**Comment by clarkmoody** [comment]
Overall, the style matches only the `0.29` changelog, with a PR number at the end of the item. Most of the rest of the entries are linked text without explicitly printing out the numbers.

File: bitcoin/CHANGELOG.md (hunk 3 above)
Comment (tcharding) [reply bitcoin/CHANGELOG.md:4]: Oh good catch, thanks, will fix.

**Comment by apoelstra** [comment]
I prefer the new style. It involves much less manual work to create.

**Comment by tcharding** [comment]
I'll fix this for now but hopefully https://github.com/rust-bitcoin/rust-bitcoin/pull/2050 merges and we never have to create changelog entries again.

**Comment by tcharding** [comment]
Rebased and dropped the `hashes` stuff, it was done already during recent `hashes` release.

**Review by apoelstra — APPROVED** [review]
ACK 24e5e19ed016476bf077f0edfecc12389b7ff5e4

**Review by clarkmoody — APPROVED** [review]
ACK 24e5e19ed016476bf077f0edfecc12389b7ff5e4

**Pull request merged**